src/lib/transcompiler.rs

```diff
@@ -2,6 +2,9 @@ use std::error::Error;
 
 use time::now_utc;
 
+extern crate urlencoding;
```

Comment:
The instruction `extern crate` is useless as you import the crate just below using `use`. The only benefit of using `extern crate` is by combining it with `#[macro_use]` at the beginning of the `main.rs` file